src/pages/admin/admin.tsx

```diff
@@ -1,26 +1,63 @@
-import React, {FC} from 'react';
+import React, {FC, Fragment, useEffect, useState} from 'react';
 import { useForm } from 'react-hook-form';
+import {Cake} from '../../App';
 
 
-const Admin: FC = () => {
-	const { register, handleSubmit } = useForm(); // initialise the hook
-	const onSubmit = (data: any)=> {
+const AdminPage: FC = () => {
+	const { register, handleSubmit } = useForm();
+	const [cakes, setCakes] = useState<Cake[]>([]);
+	const [orders, setOrders] = useState([]);
+
+	useEffect (()=>{
+		setCakes(	JSON.parse(localStorage.getItem('cakes') as string));
+	},[]);
+
+	useEffect(()=>{
+		setOrders( JSON.parse(localStorage.getItem('orders') as string));
+	}, []);
+
+	const onSubmit = (data: object)=> {
 		const cakesInLocalStorage = JSON.parse(localStorage.getItem('cakes') as string) || [];
```

Comment:
Here, you've already gotten the cakes up in the useState. you can refer to that variable here. Since we're converting this to context, this would be goten from the context too.